shell/core/plugins.js

```diff
@@ -469,3 +469,16 @@ export default function(context, inject, vueApp) {
     vueApp
   );
 }
+
+export function getProviders(context) {
+  // Custom Providers from extensions - initialize each with the store and the i18n service
+  // Wrap in try ... catch, to prevent errors in an extension breaking the page
```

Comment:
this is a nice idea, it may be best though to wrap each attempt at initialising the provider instead of all together. that way if one failed it wouldn't skip the others later in the array. this would be an improvement on the original code

Reply:
the name of the failing provider could then also be included in the console message